src/core/store/redis.ts

```diff
@@ -0,0 +1,125 @@
+import BigNumber from 'bignumber.js'
+import Redis from 'ioredis'
+import { SettlementStore } from '.'
+
+/**
+ * Redis Key Namespace
+ * =========================================
+ *
+ * accounts
+ * - Set of identifiers for all active accounts
+ *
+ * accounts:[accountId]:settlement-requests:[idempotencyKey]
+ * - Hash of each request from connector to send an outgoing settlement
+ * - `amount` -- arbitrary precision string of amount queued for settlement
+ * - `last_request_timestamp` -- UNIX timestamp in seconds when most recent request
+ *    was received with the same idempotency key
+ *
+ * accounts:[accountId]:queued-settlements
+ * - List of arbitrary precision strings of outgoing settlements to be performed (queued and failed)
+ *
+ * accounts:[accountId]:uncredited-settlements
+ * - List of arbitrary precision strings of incoming settlements yet to be credited by connector
+ */
+
+/** Configuration options for the connection to the Redis database */
+export interface RedisOpts extends Redis.RedisOptions {
+  client?: Redis.Redis
+  uri?: string
+}
+
+type RedisListTxResponse = [[null, string[]], [null, null]]
+
+// TODO Should this define a more specific "RedisSettlementStore" type?
+
+export const connectRedis = async ({ client, uri, ...opts }: RedisOpts = {}): Promise<
+  SettlementStore
+> => {
+  /**
+   * After a close reading of IORedis, options set by left params supercede
+   * options set by right params (due to the use Lodash _.defaults):
+   * https://github.com/luin/ioredis/blob/1baff479b2abfb1cba73e84ce514b3330b2b0993/lib/redis/index.ts#L193
+   */
+  const redis = client ? client : new Redis(uri, opts)
+
+  redis.defineCommand('deleteAccount', {
```

Comment:
Does this delete all past information about an account? If so, it may be worth having delete be a _soft delete_ and a _hard delete_ requires manually doing it?